### DISCUSSION TITLE
Hello , I want to ask if you could help me make tabs such as yourself for my project ?

### DISCUSSION THREAD

**Discussion by faouzikemala** [os-js/OS.js#837] [Q&A] [answered]
hello , 

Can you help me make tabs than i could drag and drop. Also help me modify it with three icons in the top corner. 
The project is a simulation of desktop software. My major problem is The DOM parts which i recreate the code with Angular ?

**Comment by andersevenrud** [comment, marked as answer]
The UI library provided does not support drag-and-drop, nor extra entries in the corner in the Tabs.

If you want this, you have to modify the provided- or make your own Tab component. 

> which i recreate the code with Angular

You can use whatever framework you want to create your own UIs, but you can't use the UI library provided with these.

I'm not familiar with Angular, so I cannot help you here, but a component like this is fairly simple and you can probably look at the OS.js sources for some pointers.

**Reply by faouzikemala** [reply]
Thanks for answering.
